lib/terraforming/resource/security_group.rb

```diff
@@ -159,7 +178,12 @@ def self_referenced_permission?(security_group, permission)
       end
 
       def security_groups
-        @client.describe_security_groups.map(&:security_groups).flatten
+        description = if @group_ids
+                        @client.describe_security_groups(group_ids: @group_ids)
+                      else
+                        @client.describe_security_groups
+                      end
```

Comment:
I opted to make this block conditional instead of doing `@client.describea_security_groups(opts)` not to break huge number of test cases :)

I can make it look something like:

```
opts = {}.tap { |o|
  o.[:group_ids] ||= @group_ids if @group_ids
}
@client.describe_security_groups(opts)
```

but I prefer separating it into another PR to ease your review.